lib/hyacinth/adapters/digital_object_search_adapter/solr/document_generator.rb

```diff
@@ -12,36 +12,45 @@ class Solr::DocumentGenerator
 
         def solr_document_for(digital_object)
           solr_document = merge_core_fields_for(digital_object)
+          merge_title_fields_for!(digital_object, solr_document)
           merge_descriptive_fields_for!(digital_object, solr_document)
           merge_rights_fields_for!(digital_object, solr_document)
-          solr_document
+          solr_document.compact
         end
 
         def merge_core_fields_for(digital_object, solr_document = {})
           merge_core_fields_for!(digital_object, solr_document.dup)
         end
 
         def merge_core_fields_for!(digital_object, solr_document = {})
-          indexable_title = digital_object.generate_title
+          indexable_identifiers = digital_object.identifiers.to_a
           solr_document.merge!(
             'id' => digital_object.uid,
             'state_ssi' => digital_object.state,
             'digital_object_type_ssi' => digital_object.digital_object_type,
             'doi_ssi' => digital_object.doi,
-            'identifier_ssim' => digital_object.identifiers.to_a,
-            'title_ss' => digital_object.generate_title,
-            'sort_title_ssi' => digital_object.generate_title(true),
+            'identifier_ssim' => indexable_identifiers.dup,
             'primary_project_ssi' => digital_object.primary_project&.string_key,
             'projects_ssim' => project_keys_for(digital_object),
-            'created_at_dtsi' => digital_object.created_at&.utc&.iso8601,
-            'updated_at_dtsi' => digital_object.updated_at&.utc&.iso8601,
+            'created_at_dtsi' => iso8601_or_nil(digital_object.created_at),
+            'updated_at_dtsi' => iso8601_or_nil(digital_object.updated_at),
             'number_of_children_isi' => digital_object.number_of_children,
             'parent_ids_ssim' => digital_object.parents.map(&:uid)
           )
+          indexable_identifiers.concat([digital_object.uid]).concat digital_object.preservation_target_uris.map { |uri| uri.split('//').last }
+          add_identifiers(indexable_identifiers, solr_document)
+        end
+
+        def merge_title_fields_for(digital_object, solr_document = {})
+          merge_descriptive_fields_for!(digital_object, solr_document.dup)
```

Comment:
Did you mean to call `merge_title_fields_for!` here?

Reply:
totally, that method just isn't called in the relevant execution path.